src/main/java/hudson/maven/reporters/TestMojo.java

```diff
@@ -167,21 +168,34 @@ public boolean canRunTests(MojoInfo mojo) {
     }
     
     @CheckForNull public Iterable<File> getReportFiles(MavenProject pom, MojoInfo mojo) throws ComponentConfigurationException {
+        File reportsDir = getReportsDirectory(pom, mojo);
+        if (reportsDir != null && reportsDir.exists())
+            return getReportFiles(reportsDir, getFileSet(reportsDir));
+        
+        return null;
+    }
+
+    /**
+     * Returns the location of test reports created by the specified MOJO.
+     * @param pom The project model.
+     * @param mojo The MOJO.
+     * @return The directory containing the test reports.
+     * @throws ComponentConfigurationException if unable to retrieve the report directory from the MOJO configuration.
+     */
+    protected File getReportsDirectory(MavenProject pom, MojoInfo mojo) throws ComponentConfigurationException {
```

Comment:
From what I see this method never returns null
Maybe makes sense to mark it as <code>@Nonnull</code> and to remove the check above